src/parse/errors.rs

```diff
@@ -904,4 +922,12 @@ impl From<fmt::Error> for Error {
     }
 }
 
-impl error::Error for Error {}
+impl error::Error for Error {
+    fn source(&self) -> Option<&(dyn error::Error + 'static)> {
+        if let Some(source) = self.source.as_deref() {
+            Some(source)
+        } else {
+            None
+        }
```

Comment:
Isn't this whole thing replaced by just `self.source.as_deref()`?

Reply:
Nope, there's a coercion here.